qemu/tests/dump_guest_core.py

```diff
@@ -0,0 +1,120 @@
+
+    def check_core_file():
+        """
+        Use gdb to check core dump file
+        """
+        process.run("echo \"source /usr/share/qemu-kvm/dump-guest-memory.py\" > %s"
+                    % gdb_command_file, shell=True)
+        process.run("echo \"set height 0\" >> %s"
+                    % gdb_command_file, shell=True)
+        if params['vm_arch_name'] == 'x86_64':
+            process.run("echo \"dump-guest-memory /tmp/vmcore X86_64\" >> %s"
+                        % gdb_command_file, shell=True)
+        if params['vm_arch_name'] == 'ppc64le':
+            process.run("echo \"dump-guest-memory /tmp/vmcore ppc64-le\" >> %s"
+                        % gdb_command_file, shell=True)
+        process.run("echo bt >> %s" % gdb_command_file, shell=True)
+        process.run("echo quit >> %s" % gdb_command_file, shell=True)
+        logging.info("gdb_command is %s" % gdb_command)
+        status, output = process.getstatusoutput(gdb_command, timeout=360)
+        os.remove(gdb_command_file)
+        os.remove(core_file)
+        logging.debug(output)
+        if status != 0:
+            test.fail("gdb command execute failed")
+        elif "<class 'gdb.MemoryError'>" in output:
+            if dump_guest_core == "on":
+                test.fail("Cannot access memory")
+
+    def check_vmcore_file():
+        """
+        Use crash to check vmcore file
+        """
+        process.run("echo bt > %s" % crash_script, shell=True)
+        process.run("echo task 0 >> %s" % crash_script, shell=True)
+        process.run("echo task 1 >> %s" % crash_script, shell=True)
+        process.run("echo quit >> %s" % crash_script, shell=True)
+        crash_cmd = "crash -i %s /usr/lib/debug/lib/modules/%s/vmlinux "
+        crash_cmd %= (crash_script, host_kernel_version)
+        crash_cmd += vmcore_file
+        status, output = process.getstatusoutput(crash_cmd, timeout=60)
+        os.remove(crash_script)
+        os.remove(vmcore_file)
+        logging.debug(output)
+        if "systemd" in output and 'swapper' in output:
+            logging.info("Crash command works as expected")
+        else:
+            test.fail("Vmcore corrupt")
+
+    # install crash/gdb/kernel-debuginfo in host
+    packages = ["crash", "gdb", "kernel-debuginfo*", "qemu-kvm-debuginfo",
+                "qemu-kvm-debugsource", "qemu-kvm-core-debuginfo"]
+    utils_package.package_install(packages)
+
+    trigger_core_dump_command = params.get("trigger_core_dump_command")
+    gdb_command = params.get("gdb_command")
+    gdb_command_file = params.get("gdb_command_file")
+    dump_guest_core = params.get("dump_guest_core")
+    pre_command = params.get("pre_command")
+    pre_command1 = params.get("pre_command1")
+    crash_script = params.get("crash_script")
+    vmcore_file = params.get("vmcore_file")
+
+    process.run(pre_command, shell=True)
+    process.run(pre_command1, shell=True)
+
+    params["extra_params"] %= dump_guest_core
```

Comment:
If you remove this one, and pre command does not need you run it by yourself, then it should be fine to remove `start_vm=no` in cfg and directly start vm with those prepared configurations.